Minecraft/src/main/java/com/microsoft/Malmo/Client/ClientStateMachine.java

```diff
@@ -1595,14 +1625,31 @@ protected void onServerTick(ServerTickEvent ev)
                 episodeHasCompleted(ClientState.WAITING_FOR_SERVER_READY);
             }
         }
-    };
+
+        @Override
+        public void onClientTick(TickEvent.ClientTickEvent ev)
+        {
+            // Check to see whether anything has caused us to abort - if so, go to the abort state.
+            if (inAbortState())
+                episodeHasCompleted(ClientState.MISSION_ABORTED);
+
+            if (++totalTicks > WAIT_MAX_TICKS)
+            {
+                String msg = "Too long waiting for world to be created.";
+                TCPUtils.Log(Level.SEVERE, msg);
+                episodeHasCompletedWithErrors(ClientState.ERROR_TIMED_OUT_WAITING_FOR_WORLD_CREATE, msg);
+            }
+        }
+    }
 
     // ---------------------------------------------------------------------------------------------------------
     /**
      * State in which an agent has finished the mission, but is waiting for the server to draw stumps.
      */
     public class MissionIdlingEpisode extends ConfigAwareStateEpisode
     {
+        int totalTicks = 0;
```

Comment:
It could be made configurable. I'll just count the ticks for now.

Reply:
Well, there's no way in here to tell whether something has gone wrong or whether it's legitimately waiting for another client - eg imagine a survival scenario where the last agent standing wins, and the last agent survives for three days after the first one died. There's no point making it configurable because there's no way to know what to configure it to!

Reply:
In a "training to deadline"  use case it could be better to just time out than to waste training time.

Reply:
In which case I think it's up to the researcher to set a timeout in their mission spec. What we definitely don't want is for Malmo to decide something is an error condition, when the mission is actually running as the researcher requested.